src/first_project_frontend/src/App.jsx

```diff
@@ -1,86 +1,58 @@
   const [age, setAge] = useState('');
-  const [showTable, setShowTable] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
 
-  const fetchMembers = async () => {
-    try {
-      const membersArray = await chapter_2.getAllMembers();
-      console.log("Members Array:", membersArray);
-      setMembers(membersArray);
-      setShowTable(true); 
-    } catch (error) {
-      console.error('Error fetching members:', error);
-    }
-  };
+  useEffect(() => {
+    const fetchMembers = async () => {
+      try {
+        const membersArray = await chapter_2.getAllMembers();
+        console.log("Members Array:", membersArray);
+        setMembers(membersArray);
+      } catch (error) {
+        console.error('Error fetching members:', error);
+      } finally {
+        setIsLoading(false);
+      }
+    };
 
+    fetchMembers();
+  }, []);
 
   const handleAddMember = async (event) => {
     event.preventDefault();
     try {
       await chapter_2.addMember({ name, age: Number(age) });
       setName('');
       setAge('');
-      fetchMembers(); 
+      const membersArray = await chapter_2.getAllMembers();
+      setMembers(membersArray);
     } catch (error) {
       console.error('Error adding member:', error);
     }
   };
 
   return (
-    <main>
-      <img src="/logo2.svg" alt="DFINITY logo" />
-      <br />
-      <br />
-      <h2>Add Member</h2>
-      <form onSubmit={handleAddMember}>
-        <label htmlFor="name">Name:</label>
-        <input
-          type="text"
-          id="name"
-          value={name}
-          onChange={(e) => setName(e.target.value)}
-          required
-        />
-        <br />
-        <label htmlFor="age">Age:</label>
-        <input
-          type="number"
-          id="age"
-          value={age}
-          onChange={(e) => setAge(e.target.value)}
-          required
-        />
-        <br />
-        <button type="submit">Add Member</button>
-      </form>
-      <br />
-      <h2>All Members</h2>
-      <button onClick={fetchMembers}>Fetch Members</button>
-      {showTable && ( 
-        <table>
-          <thead>
-            <tr>
-              <th>Name</th>
-              <th>Age</th>
-            </tr>
-          </thead>
-          <tbody>
-            {members.map((member, index) => (
-              <tr key={index}>
-                <td>{member.name}</td>
-                <td>{member.age.toString()}</td>
-              </tr>
-            ))}
-          </tbody>
-        </table>
-      )}
-    </main>
+    <Router>
+      <main>
+        <img src="/logo2.svg" alt="DFINITY logo" style={{ marginBottom: '40px' }} />
+        <Routes>
+          <Route path="/" element={<Navigate to="/members" />} />
+          <Route path="/members" element={<Members members={members} isLoading={isLoading} />} />
+          <Route path="/members/new" element={<AddMember handleAddMember={handleAddMember} name={name} setName={setName} age={age} setAge={setAge} />} />
```

Comment:
In Addmember only the handleAddmember is required , name, setName, age  etc are not required and i have removed it
For the case of  isLoading it  is used for checking condition whether the data is loaded or not 
Rather than <Members/> component this is used because the backend is called in the App.jsx|
so it is easy to acess data from backend So the data is passed with the Route